BillTrack.Persistence/Configuration/DepartmentConfiguration.cs

```diff
@@ -0,0 +1,20 @@
+using BillTrack.Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace BillTrack.Persistence.Configuration;
+
+public class DepartmentConfiguration : IEntityTypeConfiguration<Department>
+{
+    public void Configure(EntityTypeBuilder<Department> builder)
+    {
+        builder.Property(d => d.Name)
+            .HasMaxLength(100)
+            .IsRequired();
+
+        builder.HasMany(d => d.Employees)
+            .WithOne(e => e.Department)
+            .HasForeignKey(e => e.DepartmentId)
+            .OnDelete(DeleteBehavior.Cascade);
```

Comment:
delete employees if delete a department? Not an option.
We had the talk about a "soft delete". So, a delete should not be possible. Any delete attempt should throw an error.

Please, remove the `OnDelete()` clause.

Reply:
fixed